controllers/sr_controller/sr/robot/vision/vectors.py

```diff
@@ -156,6 +156,7 @@ def angle_between(vec_a: Vector, vec_b: Vector) -> float:
     dp = dot_product(vec_a, vec_b)
     mod_ab = vec_a.magnitude() * vec_b.magnitude()
     cos_theta = dp / mod_ab
+    cos_theta = min(1, max(cos_theta, -1))  # Ensures it is within the bounds of 1 and -1
```

Comment:
While this is the right domain, I'd much rather have an error than allow possible errors through.

Since we're already attempting to convert the result of the dot product into the right domain it could be that there's something deeper amiss here.

Reply:
`vec_a=Vector((-0.1757748978918965, -0.027978574520328252, -0.09121614242503248)) vec_b=Vector((0.8788744894594824, 0.13989287260164124, 0.4560807121251624))`